lib/statsd/instrument/uds_connection.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+module StatsD
+  module Instrument
+    class UdsConnection
+      DEFAULT_MAX_PACKET_SIZE = 8_192
+
+      def initialize(socket_path, max_packet_size: DEFAULT_MAX_PACKET_SIZE)
+        if max_packet_size <= 0
+          StatsD.logger.warn do
+            "[StatsD::Instrument::UdsConnection] max_packet_size must be greater than 0, " \
+              "using default: #{DEFAULT_MAX_PACKET_SIZE}"
+          end
+        end
+        @socket_path = socket_path
+        @max_packet_size = max_packet_size
+      end
+
+      def send_datagram(message)
+        socket.sendmsg_nonblock(message)
```

Comment:
The `_nonblock` version of IO methods raise an exception if the socket can't be written into and you're suppose to retrt later: https://docs.ruby-lang.org/en/3.3/BasicSocket.html#method-i-sendmsg_nonblock

So this doesn't seem correct.

If you want to just drop the packet if the socket is full, then you can dismiss the exception with `exception: false` and `:wait_writable` will be returned in case the socket is full.